src/Raven.Client/Documents/Session/DocumentSession.Attachments.cs

```diff
@@ -19,6 +21,35 @@ namespace Raven.Client.Documents.Session
     /// </summary>
     public partial class DocumentSession
     {
+        public AttachmentResult[] GetAttachmentNames<T>(string documentId)
+        {
+            var entity = Load<T>(documentId);
```

Comment:
You can look it up by id in the session, and if it isn't there, make a `Head` call, to get the metadata.